homeassistant/components/fibaro/light.py

```diff
@@ -83,93 +83,47 @@ def __init__(self, fibaro_device):
-            or supports_color
-            or supports_white_v
-        )
-
-        # Configuration can override default capability detection
-        if supports_dimming:
-            self._supported_flags |= SUPPORT_BRIGHTNESS
-        if supports_color:
-            self._supported_flags |= SUPPORT_COLOR
-        if supports_white_v:
-            self._supported_flags |= SUPPORT_WHITE_VALUE
+        supports_dimming = "levelChange" in fibaro_device.interfaces
+
+        if supports_color and supports_white_v:
+            self._attr_supported_color_modes = {COLOR_MODE_RGBW}
+            self._attr_color_mode = COLOR_MODE_RGBW
+        elif supports_color:
+            self._attr_supported_color_modes = {COLOR_MODE_RGB}
+            self._attr_color_mode = COLOR_MODE_RGB
+        elif supports_dimming:
+            self._attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS}
+            self._attr_color_mode = COLOR_MODE_BRIGHTNESS
+        else:
+            self._attr_supported_color_modes = {COLOR_MODE_ONOFF}
+            self._attr_color_mode = COLOR_MODE_ONOFF
 
         super().__init__(fibaro_device)
         self.entity_id = ENTITY_ID_FORMAT.format(self.ha_id)
 
-    @property
-    def brightness(self):
-        """Return the brightness of the light."""
-        return scaleto255(self._brightness)
-
-    @property
-    def hs_color(self):
-        """Return the color of the light."""
-        return self._color
-
-    @property
-    def white_value(self):
-        """Return the white value of this light between 0..255."""
-        return self._white
-
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        return self._supported_flags
-
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
         async with self._update_lock:
             await self.hass.async_add_executor_job(partial(self._turn_on, **kwargs))
 
     def _turn_on(self, **kwargs):
         """Really turn the light on."""
-        if self._supported_flags & SUPPORT_BRIGHTNESS:
-            target_brightness = kwargs.get(ATTR_BRIGHTNESS)
-
-            # No brightness specified, so we either restore it to
-            # last brightness or switch it on at maximum level
-            if target_brightness is None:
-                if self._brightness == 0:
-                    if self._last_brightness:
-                        self._brightness = self._last_brightness
-                    else:
-                        self._brightness = 100
-            else:
-                # We set it to the target brightness and turn it on
-                self._brightness = scaleto100(target_brightness)
-
-        if self._supported_flags & SUPPORT_COLOR and (
-            kwargs.get(ATTR_WHITE_VALUE) is not None
-            or kwargs.get(ATTR_HS_COLOR) is not None
+        if brightness_supported(self.supported_color_modes) and (
+            ATTR_BRIGHTNESS in kwargs
         ):
```

Comment:
It's actually not needed to guard; the base entity will not pass in parameters not supported by the light.
```suggestion
        if ATTR_BRIGHTNESS in kwargs:
```